crates/esplora/src/async_ext.rs

```diff
@@ -205,12 +205,10 @@ async fn fetch_block<S: Sleeper>(
 
     // We avoid fetching blocks higher than previously fetched `latest_blocks` as the local chain
     // tip is used to signal for the last-synced-up-to-height.
-    let &tip_height = latest_blocks
-        .keys()
-        .last()
-        .expect("must have atleast one entry");
-    if height > tip_height {
-        return Ok(None);
+    if let Some(tip_height) = latest_blocks.keys().last().copied() {
+        if height > tip_height {
+            return Ok(None);
+        }
     }
```

Comment:
I think it's better to return `None` if there are no `latest_blocks` as the `latest_blocks` protects us from creating a chain update with inconsistent state. 

However, `latest_blocks` should really never be empty - how about we put a `debug_assert!` here?